### PR TITLE
DEV-2943 | Fix bug with restoring heroku db backup locally

### PR DESCRIPTION
**Pull request by benjaminEwhite**
Please fill out each section even if it's just with "N/A"

#### What's this PR do?

This PR fixes a (sort of) silent error that occurred when restoring Heroku Postgres backups locally.  The error caused the `config_denylistword` table to not be restored, even though running `python manage.py showmigrations` would show that table as having been created (insofar as it shows the migration creating it ran).

The underlying problem here turns out to be that Heroku installs the ci_text extension for us, and there’s some sort of funkiness (yes, it’s a technical term) around restoring Heroku Postgres backup locally where by ci_text is not found. Here’s relevant bit from output of logs of verbose pg_restore:

```
pg_restore: creating TABLE "public.config_denylistword"
pg_restore: from TOC entry 219; 1259 7441936 TABLE config_denylistword ecrnjtehuqbzyu
pg_restore: error: could not execute query: ERROR:  schema "heroku_ext" does not exist
LINE 3:     "word" "heroku_ext"."citext" NOT NULL
                   ^
Command was: CREATE TABLE "public"."config_denylistword" (
    "id" integer NOT NULL,
    "word" "heroku_ext"."citext" NOT NULL
);
```

The fix is to manually create the extension ahead of restoring backup ([a la this SO answer](https://stackoverflow.com/a/73557740)).

As a long term fix,  this PR adds two make commands and updates db restore instructions in readme. This necessitated a small change to `.envrc.example` (which in turn should be copied over to `.envrc` in local dev environments).

#### Why are we doing this? How does it help us?

Acute problem for me was that I was unable to edit or create revenue programs in the Django admin when running locally, which blocked me on a ticket I was working on. Underlying problem was that Django needed config_denylistword table to be in place because RP links to it.

This helps us by giving other devs away to avoid the underlying bug, and its downstream side effects.

#### How should this be manually tested? Please include detailed step-by-step instructions.

Testing this is for devs only.

1. Update your local `.envrc` file in line with the changes in `.envrc.example` from this PR.
2. Follow the backup capture and backup restore instructions from the updated README file for this app.
3. Use pg_admin or similar to connect to the local db. You should see that the `config_denylistword` table has been created.
4. Additionally, if you inspect the logs from running the `make restore-db-backup` command, you should not see an error that looks like this:

```
pg_restore: creating TABLE "public.config_denylistword"
pg_restore: from TOC entry 219; 1259 7441936 TABLE config_denylistword ecrnjtehuqbzyu
pg_restore: error: could not execute query: ERROR:  schema "heroku_ext" does not exist
LINE 3:     "word" "heroku_ext"."citext" NOT NULL
                   ^
Command was: CREATE TABLE "public"."config_denylistword" (
    "id" integer NOT NULL,
    "word" "heroku_ext"."citext" NOT NULL
);
```

#### Did this PR make changes to the user interface that may require changes to the user-facing docs?

No

#### Have automated unit tests been added? If not, why?

No. Not really something we test.

#### How should this change be communicated to end users?

N/A

#### Are there any smells or added technical debt to note?

No

#### Has this been documented? If so, where?

Yes, in README.md

#### What are the relevant tickets? Add a link to any relevant ones.

[DEV-2943](https://news-revenue-hub.atlassian.net/browse/DEV-2943)

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?

Devs will need to update their `.envrc` file in line with the updated `.envrc.example` file in order for the backup restore command to work.

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:

No

[DEV-2943]: https://news-revenue-hub.atlassian.net/browse/DEV-2943?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by leo-dcfa — APPROVED** [review]
I used the scripts in this ticket and the `config_denylistword` table error disappeared. LGTM.

File: README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Revenue Engine <!-- omit in toc -->
+****# Revenue Engine <!-- omit in toc -->
```
Comment (benjaminEwhite) [inline README.md:1]:
```suggestion
# Revenue Engine <!-- omit in toc -->
```

File: .envrc.example [hunk 2]
```diff
@@ -3,7 +3,8 @@ export REACT_APP_HUB_STRIPE_API_PUB_KEY=pk_test_???
 export REACT_APP_CAPTURE_PAGE_SCREENSHOT=true
 export STRIPE_WEBHOOK_SECRET=whsec_???
 export DJANGO_SETTINGS_MODULE=revengine.settings.local
-export DATABASE_URL=postgres://postgres@127.0.0.1:54000/revengine
+export DATABASE_ROOT=postgres://postgres@127.0.0.1:54000
```
Comment (x110dc) [inline .envrc.example:6]: i don't see DATABASE_ROOT used anywhere else except the next line, so not sure what the value of adding an extra variable is here.

File: README.md [hunk 3]
```diff
@@ -363,11 +363,17 @@ heroku pg:backups:download --app rev-engine-test b001
...
-NOTE: The `pg_restore` command assumes that you can run `$psql` and get a prompt for your local database.
+NOTE: The Make commands below assume that you can run `$psql` and get a prompt for your local database.
+
+
 
 ```sh
-heroku pg:backups:download --app rev-engine-test b001
-pg_restore --verbose --clean --no-acl --no-owner -d revengine latest.dump
+# assuming that you already have a local backup up and running. if not, skip the first two commands
+make reset-db
```
Comment (x110dc) [inline README.md:372]: I don't see `reset-db` anywhere. Should this be `drop-db`?

File: README.md [hunk 4]
```diff
@@ -363,11 +363,17 @@ heroku pg:backups:download --app rev-engine-test b001
 
 #### Restore backup
 
-NOTE: The `pg_restore` command assumes that you can run `$psql` and get a prompt for your local database.
+NOTE: The Make commands below assume that you can run `$psql` and get a prompt for your local database.
+
```
Comment (x110dc) [inline README.md:367]: `psql` not `$psql`, I think.

**Review by x110dc — CHANGES_REQUESTED** [review]

File: .envrc.example (hunk 2 above)
Comment (benjaminEwhite) [reply .envrc.example]: Removed in 10ddf5cbe04

File: README.md (hunk 4 above)
Comment (benjaminEwhite) [reply README.md]: b688d6540684

File: README.md (hunk 3 above)
Comment (benjaminEwhite) [reply README.md]: 8cf8e29418b

**Review by x110dc — APPROVED** [review]
LGTM.

**Review by x110dc — APPROVED** [review]
LGTM.

**Pull request merged**